src/main/java/org/apache/commons/net/util/SubnetUtils.java

```diff
@@ -116,96 +82,232 @@ public boolean isInclusiveHostCount() {
      * @since 2.2
      */
     public void setInclusiveHostCount(boolean inclusiveHostCount) {
-        this.inclusiveHostCount = inclusiveHostCount;
+        subnetInfo.setInclusiveHostCount(inclusiveHostCount);
     }
 
+    /**
+     * Creates subnet summary information based on the provided IPv4 or IPv6 address in CIDR-notation,
+     * e.g. "192.168.0.1/16" or "2001:db8:0:0:0:ff00:42:8329/46"
+     *
+     * NOTE: IPv6 address does NOT allow to omit consecutive sections of zeros in the current version.
+     *
+     * @param cidrNotation IPv4 or IPv6 address
+     * @return a SubnetInfo object created from the IP address.
+     * @since 3.7
+     */
+    public static SubnetInfo getByCIDRNortation(String cidrNotation) {
+        if (Pattern.matches(IPV4_ADDRESS, cidrNotation)) {
+            return new IP4Subnet(cidrNotation);
+        } else if (Pattern.matches(IPV6_ADDRESS, cidrNotation)) {
+            return new IP6Subnet(cidrNotation);
+        } else {
+            throw new IllegalArgumentException("Could not parse [" + cidrNotation + "]");
+        }
+    }
 
+    /**
+     * Creates IPv4 subnet summary information, given a dotted decimal address and mask.
+     *
+     * @param address an IP address, e.g. "192.168.0.1"
+     * @param mask a dotted decimal netmask e.g. "255.255.0.0"
+     * @throws IllegalArgumentException if the address or mask is invalid,
+     * e.g. the address does not match n.n.n.n where n=1-3 decimal digits, or
+     * the mask does not match n.n.n.n which n={0, 128, 192, 224, 240, 248, 252, 254, 255}
+     * and after the 0-field, it is all zeros.
+     * @return an IP4Subnet object generated based on <code>address</code> and <code>mask</code>.
+     * @since 3.7
+     */
+    public static IP4Subnet getByMask(String address, String mask) {
+        return new IP4Subnet(address, mask);
+    }
 
     /**
      * Convenience container for subnet summary information.
-     *
+     * @since 3.7 (reorganized)
```

Comment:
@since markers are only used for new methods, fields, classes that are part of the public API